docs/SIGNAL_ROADMAP.MD

```diff
@@ -1,5 +1,56 @@
+- **Keine Servo-Unterstützung:** Die Architektur ist ausschließlich auf Lichtsignale (NeoPixel) ausgelegt. Die Ansteuerung von mechanischen Signalen über Servos ist nicht vorgesehen.
+- **Abhängigkeit von CV-Programmierung:** Die Konfiguration erfolgt ausschließlich über DCC-CVs. Dies ist umständlich, schlecht dokumentiert und für Einsteiger nur schwer zugänglich. Es fehlt eine grafische Konfigurationsoberfläche.
+- **Fehlende Bidirektionale Kommunikation:** Der Decoder unterstützt kein RailCom, was eine Auslesung von CVs "on-the-main" (PoM) oder eine anlagenweite Zustandsrückmeldung unmöglich macht.
+- **Mangelnde Test-Automatisierung:** Unit-Tests sind nur rudimentär vorhanden. Dies erhöht das Risiko von Regressionen bei jeder Code-Änderung.
+
+### Strategische Feature-Roadmap (Top 20)
+Die folgende Liste ist priorisiert und repräsentiert die wichtigsten Entwicklungsschritte zur Transformation des Decoders.
+
+**Phase 1: Architektur-Modernisierung & Kern-Features**
+1.  [ ] **Dynamischer Signal-Dispatcher:** Entwicklung eines zentralen Dispatchers, der Signal-Objekte zur Laufzeit basierend auf CV-Werten instanziiert. Ermöglicht mehrere, verschiedene Signaltypen pro Decoder.
+2.  [ ] **Datengetriebene Signal-Definitionen (XML/JSON):** Verlagerung der Signallogik (Aspekte, Zustände, Regeln) aus dem C++ Code in Konfigurationsdateien, die vom Decoder geparst werden.
+3.  [ ] **Servo-Subsystem-Integration:** Implementierung einer robusten Servo-Ansteuerungsklasse, inklusive Endlagen-Justierung, Geschwindigkeitssteuerung und sanftem Anlauf/Stopp.
+4.  [ ] **Erweiterung für mechanische Signale:** Definition und Implementierung der ersten mechanischen Signale (z.B. deutsches `Hp`-Formsignal) unter Nutzung des Servo-Subsystems.
+5.  [ ] **Umfassendes Unit-Testing-Framework:** Aufbau einer soliden Test-Suite (z.B. mit GoogleTest) für die Kern-Logik (DCC-Parser, Signal-Dispatcher, etc.), um Stabilität zu gewährleisten.
+
+**Phase 2: Konnektivität & Benutzererfahrung (UX)**
+6.  [ ] **Integriertes Web-Interface:** Ein auf dem Decoder laufender Webserver zur grafischen Konfiguration aller CVs und Decoder-Optionen über den Browser.
+7.  [ ] **WLAN-Konnektivität (On-Board):** Nutzung eines Mikrocontrollers mit WLAN (z.B. ESP32) oder eines Co-Prozessors, um das Web-Interface bereitzustellen.
+8.  [ ] **RailCom-Implementierung (Cutout & Channel 1):** Basis-Implementierung von RailCom zur Übermittlung der Decoder-Adresse und für PoM-CV-Lesen.
+9.  [ ] **Live-Vorschau im Web-Interface:** Grafische Darstellung des konfigurierten Signals und seiner Aspekte direkt im Browser, analog zum bestehenden Web-Tool.
+10. [ ] **Firmware-Updates über Web-Interface (OTA):** Ermöglichen von "Over-the-Air"-Firmware-Updates, ohne dass der Decoder ausgebaut werden muss.
```

Comment:
OTA sollte "Over The Rail (OTR)" heissen, wir möchten also die Programmierung auf dem Gleis unterstützen.